zeroconf/__init__.py

```diff
@@ -1281,20 +1281,19 @@ def remove_key(cache: dict, key: str, entry: DNSRecord) -> None:
     def get(self, entry: DNSEntry) -> Optional[DNSRecord]:
         """Gets an entry by key.  Will return None if there is no
         matching entry."""
-        try:
-            list_ = self.cache[entry.key]
-            for cached_entry in reversed(list_):
-                if entry.__eq__(cached_entry):
-                    return cached_entry
-            return None
-        except (KeyError, ValueError):
-            return None
+        for cached_entry in reversed(self.entries_with_name(entry.key)):
+            if entry.__eq__(cached_entry):
+                return cached_entry
+        return None
```

Comment:
@bdraco Let's, going forward, change only one logical thing in a single PR and not bundle fixes with improvements if they're not closely related.

Nice PR though!

Reply:
You would think that after splitting up https://github.com/jstasiak/python-zeroconf/pull/361 I would have noticed this on the second pass when I wrote the change text and split it again, but 🤦